### ISSUE TITLE
Bad results on values larger than 2^31

### ISSUE THREAD

**Issue by fireice-uk** [biocore-ntnu/ncls#38]
```
from ncls import NCLS
import numpy as np

ncls = NCLS(np.array([1000000000, 1010000000], dtype=np.int64), np.array([1100000000, 1020000000], dtype=np.int64), np.array([42, 43], dtype=np.int64))

it = ncls.find_overlap(1010000001, 1010000002)
for i in it:
    print(i)

ncls = NCLS(np.array([4000000000, 4010000000], dtype=np.int64), np.array([4100000000, 4020000000], dtype=np.int64), np.array([42, 43], dtype=np.int64))

it = ncls.find_overlap(4010000001, 4010000002)
for i in it:
    print(i)
```

Output:

```
(1000000000, 1100000000, 42)
(1010000000, 1020000000, 43)
(4000000000, 4100000000, 42)
```

**Comment by endrebak** [comment]
Thanks for reporting. Will look at it :)

**Comment by endrebak** [comment]
This also happens in find_all_overlaps_both, which leads me to believe the error is in [this function](https://github.com/biocore-ntnu/ncls/blob/master/ncls/src/intervaldb.c#L430).

```
>>> ncls.all_overlaps_both(np.array([4010000001], dtype=np.int64), np.array([4010000002], dtype=np.int64), np.array([0], dtype=np.int64))
(array([0]), array([42]))
```

I've always used Linux for testing, but now I only have access to a mac. Which system are you on?

**Comment by endrebak** [comment]
It should be fixed in 0.0.54. I am grateful for the report!
